lib/util.js

```diff
@@ -72,6 +71,79 @@ const errorToString = Error.prototype.toString;
 var CIRCULAR_ERROR_MESSAGE;
 var Debug;
 
+// eslint-disable-next-line
+const escapeSequencesRegExp = /[\\'\x00-\x1f]/g;
+const keyStrRegExp = /^[a-zA-Z_][a-zA-Z_0-9]*$/;
+const textRegExp = /^[a-zA-Z_0-9 ,.?!+\-&"öäüÜÖÄß():;<>{}[\]=$]*$/;
```

Comment:
Where did this regexp come from?

Reply:
I came up with it. I tried to think of any "somewhat common" characters that do not have to be escaped but might be used anyway. I benchmarked the result performance and it does not really change from using `/^[a-zA-Z_0-9 ,.?!+\-&"öäüÜÖÄß():;<>{}[\]=$]*$/` to e.g. `/^[a-zA-Z_0-9 ,.?!+\-$]*$/`

Reply:
All unicode characters can be inserted unescaped for example, that's why I was confused about the inclusion of only a handful of them. www.json.org shows the characters allowed in a string literal. I think the list of characters can be shortened/simplified.

Reply:
Originally I used `escapeSequencesRegExp` but it turned out to be slower than using `textRegExp`. As it is somewhat about common cases I used this one. I also tried to point out that performance wise it does not make any difference if the RegExp is further simplified.

Reply:
FWIW I just ran some quick benchmarks with simple inputs and using a simpler, negated regexp check (e.g. `/[\x00-\x1F\x7F']/`) seems to be noticeably quicker for strings containing a character needing to be escaped. For strings that don't need escaping, a simple loop over the string was measurably faster, but the performance difference there is not as large as the negated regexp in the other case.

Reply:
Hm, yes. Somehow I messed up my benchmark about only checking the escape sequences. I only added that extra RegExp because I was certain my benchmark results were better for the individual one but when benchmarking it again I confirm your results.

However, I do remember why I used a RegExp in general and not use the loop as default: the performance is somewhat consistent with the RegExp but with the loop it becomes more and more expensive in case there are characters that have to be escaped. I will check if I find a alternative way way to get the common case even better.

Reply:
I realized that I made a mistake while originally benchmarking the RegExp and therefore I had the impression the individual one would be faster.

I also had another look at the loop against replace and I increased the switch to 100 chars. It seems to be the better break even point.